pandas/core/nanops.py

```diff
@@ -102,26 +102,14 @@ def __call__(self, alt):
 
         @functools.wraps(alt)
         def f(values, axis=None, skipna=True, **kwds):
+            min_count = kwds.get('min_count', 1)
             if len(self.kwargs) > 0:
                 for k, v in compat.iteritems(self.kwargs):
                     if k not in kwds:
                         kwds[k] = v
             try:
-                if values.size == 0:
-
-                    # we either return np.nan or pd.NaT
-                    if is_numeric_dtype(values):
-                        values = values.astype('float64')
-                    fill_value = na_value_for_dtype(values.dtype)
-
-                    if values.ndim == 1:
-                        return fill_value
-                    else:
-                        result_shape = (values.shape[:axis] +
-                                        values.shape[axis + 1:])
-                        result = np.empty(result_shape, dtype=values.dtype)
-                        result.fill(fill_value)
-                        return result
+                if values.size < min_count:
```

Comment:
values could be 2d here, so in theory should handle this on a per-column basis (but not 100% sure if it is ever practically 2d).